foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/descriptors/RecordInstantiationPolicy.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Oracle - initial API and implementation from Oracle TopLink
+package org.eclipse.persistence.internal.descriptors;
+
+import org.eclipse.persistence.descriptors.ClassDescriptor;
+import org.eclipse.persistence.exceptions.DescriptorException;
+import org.eclipse.persistence.internal.sessions.AbstractRecord;
+import org.eclipse.persistence.internal.sessions.AbstractSession;
+import org.eclipse.persistence.mappings.AggregateObjectMapping;
+import org.eclipse.persistence.mappings.DatabaseMapping;
+import org.eclipse.persistence.mappings.DirectToFieldMapping;
+import org.eclipse.persistence.queries.ReadObjectQuery;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.RecordComponent;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <b>Purpose</b>: Allows customization of how an {@code java.lang.Record} is created/instantiated.<p>
+ */
+public class RecordInstantiationPolicy extends InstantiationPolicy {
+
+
+    /**
+     * Default constructor
+     */
+    public RecordInstantiationPolicy() {
+        super();
+    }
+
+    /**
+     * Build and return a new instance, using the appropriate mechanism.
+     */
+    @Override
+    public Object buildNewInstance() throws DescriptorException {
+        return null;
+    }
+
+    public Object buildNewRecordInstance(Class<Record> clazz, List<DatabaseMapping> mappings, AbstractRecord databaseRow, AbstractSession session) {
```

Comment:
this does not look right to me - this method should be private (max protected) and called by `buildNewInstance`